scipio/src/io/dma_file.rs

```diff
@@ -462,6 +434,79 @@ impl DmaFile {
     }
 }
 
+/// A wrapper over `std::fs::File` which carries a path (for better error
+/// messages) and prints a warning if closed synchronously.
+#[derive(Debug)]
+struct EnhancedFile {
```

Comment:
I am not thrilled about the name tbh. I *think* I would like `ScipioFile` more, but it seems that we generally don't use "Scipio" prefix anywhere else?

Reply:
`ScipioFile` is fine as long as it doesn't get confusing.

`Directory` is well, a directory.
`DmaFile` allows for dma operations (O_DIRECT) randomly issued at direct positions.
`StreamReader` and `StreamWriter` allow for linear access (currently only over O_DIRECT).
I have a queued patchset for `BufferedFile` which will use buffered I/O for the simple applications that don't want to deal with O_DIRECT.

So I wouldn't be opposed to `ScipioFile` as something common to *all* of those.

Closing seems like something that will be shared between `Directory`, `BufferedFile` and `DmaFile`. It's fair to say that the streams are not *really* files, but rather abstractions on top of file.

So maybe?